jsonschema.go

```diff
@@ -92,16 +96,33 @@ func checkString(rawdata interface{}, format interface{}) error {
 	return nil
 }
 
-func checkFloat(rawdata interface{}, format interface{}) error {
-	_, ok := rawdata.(float64)
+func checkNumber(rawdata interface{}) (json.Number, error) {
+	res, ok := rawdata.(json.Number)
 	if !ok {
-		return fmt.Errorf("expected float, got [%s]", reflect.TypeOf(rawdata))
+		return res, fmt.Errorf("expected json number, got [%s]", reflect.TypeOf(rawdata))
+	}
+	return res, nil
+}
+
+func checkFloat(rawdata interface{}, format interface{}) error {
+	number, err := checkNumber(rawdata)
+	if err != nil {
+		return err
+	}
+	if !strings.Contains(number.String(), ".") {
+		return fmt.Errorf("expected float number, got int[%s]", number)
 	}
 	return nil
 }
 
 func checkInt(rawdata interface{}, format interface{}) error {
-	// TODO
+	number, err := checkNumber(rawdata)
+	if err != nil {
+		return err
+	}
+	if strings.Contains(number.String(), ".") {
```

Comment:
here parsing with Int64() will be better, for some reason I thought that the method would try to coerce the float to an int, rounding it up, but it won't. Thanks